share/wake/lib/core/list.wake

```diff
@@ -67,14 +67,26 @@ export def tail: List a => List a = match _
     Nil  = Nil
     _, t = t
 
-# map: create a new List by applying a function f to each element of a List.
-#
-#   map str    (seq 5) = "0", "1", "2", "3", "4", Nil
-#   map (_+10) (seq 5) = 10, 11, 12, 13, 14, Nil
-export def map (f: a => b): List a => List b =
-    def loop = match _
+# Create a new List by applying the function *mapFn* to each element of *list*.
+# The *map* function (along with *foldl*) is generally how one implements loops in wake.
+# This function (like most in wake) runs *mapFn* in parallel.
+#
+# Parameters:
+#  - *mapFn*: The function to apply to each element
+#  - *list*: The List of elements to feed to *mapFn*
+#  - returns: A new List with the outputs of *mapFn*
```

Comment:
I normally see return values documented separately from the parameters, and oftentimes not really documented in a way that is separate from the main description.

I couldn't find many examples of this in other languages, since most go with the directive-style thing where you add something like `@param`, `:param`, or `\param` for each parameter, and have a separate directive for `@returns`. Swift has a Markdown-style format where they have [separate Parameters and Returns sections](https://github.com/apple/swift/blob/3680f9c2602e7a8c6504f512c017922eaa20ffc2/docs/DocumentationComments.md#returns-field).

Reply:
We could just not have a returns. It is usually pretty redundant with the function description.

Reply:
Giving it a section entirely for `return` wastes too much of the precious tooltip area IMO.

Reply:
A description of the returned values can occasionally be helpful, particularly in cases where we're returning a `Pair` or `Triple`; consider `splitBy` where it would probably be best to explicitly say that the `True` values are the first element.  In most cases, though, I agree that it's not required, and maybe not even desirable.

Reply:
Ah. In that case, I think a full section like for Parameters might be justified, along the lines of:

```
# Returns `Pair true false`, where:
#  - `true`: List of the elements for which `acceptFn` returned `True`
#  - `false`: List of the elements for which `acceptFn` returned `False
```

Reply:
I do agree that's probably the best option, though it's heavyweight enough that we will want to be a bit careful about where we use it.

Reply:
Sounds like we have a consensus, then. Add a "Returns:" section with multiple bullet points only when it's warranted.